### PR TITLE
allow URLs without template for downloading git repos

### PR DESCRIPTION
**Pull request by tgummerer**
For remote plugins we want to allow URLs without a scheme to be used for specifying the plugin location.  We want to be consistent with this for example for downloading templates in pulumi new.

Allow this parsing of Git URLs in the git utility.

### CODE REVIEW

File: sdk/go/common/util/gitutil/git.go [hunk 1]
```diff
@@ -650,6 +650,14 @@ type gitRepoURLParts struct {
 }
 
 func parseGitRepoURLParts(rawurl string) (gitRepoURLParts, error) {
+	urlRegex := regexp.MustCompile(`^[^\./][a-zA-Z0-9-]*\.[a-z]+/[a-zA-Z0-9-/]+$`)
+	if urlRegex.MatchString(rawurl) {
+		// We want to allow "naked" URLs, such as github.com/pulumi/pulumi-provider in addition to
+		// full URLs such as https://github.com/pulumi/pulumi-provider for convenience.  Prefix
+		// https:// to these URLs, as we assume that protocol.
+		fmt.Println("matched", rawurl)
```
Comment (Frassle) [inline sdk/go/common/util/gitutil/git.go:658]:
```suggestion
```

File: sdk/go/common/util/gitutil/git.go [hunk 2]
```diff
@@ -650,6 +650,14 @@ type gitRepoURLParts struct {
 }
 
 func parseGitRepoURLParts(rawurl string) (gitRepoURLParts, error) {
+	urlRegex := regexp.MustCompile(`^[^\./][a-zA-Z0-9-]*\.[a-z]+/[a-zA-Z0-9-/]+$`)
+	if urlRegex.MatchString(rawurl) {
```
Comment (Frassle) [inline sdk/go/common/util/gitutil/git.go:654]: Can we not just check that there _isn't_ a scheme at the start of the url already? That's probably a simpler regex.

File: sdk/go/common/util/gitutil/git.go (hunk 1 above)
Comment (tgummerer) [reply sdk/go/common/util/gitutil/git.go:658]: Whoops, thanks!

File: sdk/go/common/util/gitutil/git.go (hunk 2 above)
Comment (tgummerer) [reply sdk/go/common/util/gitutil/git.go:654]: Yeah that's probably good enough actually.  I was worried about still allowing local repos working, but thinking again, we actually never allowed that, so there's no point in worrying about those (and I don't think most people would want that anyway)

**Review by Frassle — APPROVED** [review]

**Pull request merged**